src/helpers/xls-reader.ts

```diff
@@ -23,15 +23,26 @@ export const convertToJson = (csv: string): Table => {
     return { data: result, schema: headers };
 };
 
-export const onLoad = async (evt: ProgressEvent<FileReader>, file: Blob, callback: (dataSourceId: string, table: Table, info: any) => void) => {// evt = on_file_select event
+export const onLoad = async (
+    dashboardId: string,
+    evt: ProgressEvent<FileReader>,
+    file: File,
+    onReadingSucceed: (table: Table, filename: string) => void,
+    onReadingEnd: () => void
+) => {// evt = on_file_select event
     const workBook = getWorkBook(evt);
     const workSheetName = workBook.SheetNames[0];/* Get first worksheet */
     const workSheet = workBook.Sheets[workSheetName];
     const data = XLSX.utils.sheet_to_csv(workSheet);/* Convert array of arrays */
 
     const table: Table = convertToJson(data);
-    const dataSourceId = await excelCommunicator.addExcelDataSource({ table, dashboardId: "bla", displayName: "blas" });
-    callback(dataSourceId, table, file);
+    try {
+        onReadingSucceed(table, file.name);
+    } catch {
+        notifyError("we couldnt add this excel...");
```

Comment:
Lets have a convention that toasters will always start the text with a capital letter